lib/phoenix/presence.ex

```diff
@@ -343,26 +354,182 @@ defmodule Phoenix.Presence do
       pubsub_server =
         opts[:pubsub_server] || raise "use Phoenix.Presence expects :pubsub_server to be given"
 
-      Phoenix.Tracker.start_link(__MODULE__, {module, task_supervisor, pubsub_server}, opts)
+      Phoenix.Tracker.start_link(
+        __MODULE__,
+        {module, task_supervisor, pubsub_server},
+        opts
+      )
     end
 
-    def init(state) do
-      {:ok, state}
+    def init({module, task_supervisor, pubsub_server}) do
```

Comment:
Since we have quite a bit of implementation here now, we should delegate back out of the injected callbacks (`init`, `handle_diff`) and all of the private functions into the `Phoenix.Presence` module. That way we aren't injecting all this code into the users module. For example, init could look like: 

`def init(state), do: Phoenix.Presence.__init__(state)` and so on.

In general in a `__using__` block, we should inject as little as code as possible. Also, I was guilty of violating this principle in the current/original `handle_diff` :)

Make sense?